detection_rules/rule_formatter.py

```diff
@@ -241,9 +248,17 @@ def _do_write(_data, _contents):
         if query:
             top.update({'query': "XXxXX"})
 
+        if threat_query:
+            top.update({'threat_query': "XXxXX"})
+
         top.update(bottom)
         top = toml.dumps(OrderedDict({data: top}), encoder=encoder)
 
+        # we want to preserve the query format, but want to modify it in the context of encoded dump
+        if threat_query:
+            formatted_threat_query = "\nthreat_query = '''\n{}\n'''{}".format(threat_query, '\n\n' if bottom else '')
+            top = top.replace('threat_query = "XXxXX"', formatted_threat_query)
```

Comment:
Do you have an example for this?

Reply:
Yes, a rule without this formatting will appear like this, invalid toml string:

```toml
threat_query = '@timestamp >= "now-24h"
'
```

with the formatting it will appear like this:

```toml
threat_query = '''
@timestamp >= "now-24h"
'''

```

<details><summary>Example Rule File</summary>
<p>

```toml
[metadata]
creation_date = "2024/08/02"
maturity = "development"
updated_date = "2024/08/02"

[rule]
author = ["INFOSEC_TEST"]
description = "Test Threat Query"
enabled = true
from = "now-5400s"
index = ["logs-okta*"]
interval = "1h"
language = "kuery"
max_signals = 100
name = "Test User Threat Query"
note = "Test threat query."
references = ["https://github.com/elastic"]
revision = 28
risk_score = 73
rule_id = "58be7d5b-f8bf-4666-83c6-251ceefea562"
severity = "high"
tags = ["Okta"]
threat_index = ["test_cluster:okta_users"]
threat_indicator_path = "threat.indicator"
threat_language = "kuery"
timestamp_override = "event.ingested"
to = "now"
type = "threat_match"
version = 24

query = '''
user.email:*
'''



threat_query = '''
@timestamp >= "now-24h"
'''


[[rule.required_fields]]
ecs = true
name = "user.email"
type = "keyword"

[[rule.threat_mapping]]

[[rule.threat_mapping.entries]]
field = "user.email"
type = "mapping"
value = "user.email"



```

</p>
</details> 
Ndjson version of above rule:
[test_threat_query_rule.ndjson.txt](https://github.com/user-attachments/files/16475132/test_threat_query_rule.ndjson.txt)